lib/dependencies.sh

```diff
@@ -113,21 +113,48 @@ yarn_prune_devdependencies() {
   fi
 }
 
+should_use_npm_ci() {
+  local build_dir=${1:-}
+  local npm_version
+
+  npm_version=$(npm --version)
+  # major_string will be ex: "4." "5." "10"
+  local major_string=${npm_version:0:2}
+  # strip any "."s from major_string
+  local major=${major_string//.}
+
+  # We should only run `npm ci` if all of the manifest files are there, and we are running at least npm 6.x
+  # `npm ci` was introduced in the 5.x line in 5.7.0, but this sees very little usage, < 5% of builds
+  if [[ -f "$build_dir/package.json" ]] && [[ -f "$build_dir/package-lock.json" ]] && (( major >= 6 )); then
+    echo "true"
+  else
+    echo "false"
+  fi
+}
+
 npm_node_modules() {
   local build_dir=${1:-}
   local production=${NPM_CONFIG_PRODUCTION:-false}
 
   if [ -e "$build_dir/package.json" ]; then
     cd "$build_dir" || return
 
-    if [ -e "$build_dir/package-lock.json" ]; then
-      echo "Installing node modules (package.json + package-lock)"
-    elif [ -e "$build_dir/npm-shrinkwrap.json" ]; then
-      echo "Installing node modules (package.json + shrinkwrap)"
+    if [[ "$(experiments_get "use-npm-ci")" == "true" ]] && [[ "$(should_use_npm_ci "$build_dir")" == "true" ]]; then
+      meta_set "supports-npm-ci" "true"
+      cd "$build_dir" || return
```

Comment:
what's the purpose of this `cd` call? could you add a comment explaining why you're cwd-ing here, but not for the old method?

the same `cd` also happens further up, is this duplication intentional?

Reply:
Duplication isn't intentional, but an artifact of moving things around. I'll delete it.

I am not sure why the `cd` command exists here, but have been intimidated to remove it. One issue could be users changing the directory in the `heroku-prebuild` script 🤔